src/components/infiniteScroll/index.js

```diff
@@ -1,35 +1,224 @@
-import React, { Component } from 'react';
+// @flow
+import * as React from 'react';
+import { fetchMoreOnInfiniteScrollLoad } from './tallViewports';
 
-export const withInfiniteScroll = Comp => {
-  return class InfiniteScroll extends Component {
-    componentDidMount() {
-      // if (!this.scrollBody) return;
-      // let node = this.scrollBody;
-      // node.addEventListener('scroll', this.handleScroll)
+type Props = {
+  element: string,
+  hasMore: boolean,
+  initialLoad: boolean,
+  loader: React.Node,
+  loadMore: Function,
+  isLoadingMore: boolean,
+  pageStart: number,
+  threshold: number,
+  useWindow: boolean,
+  isReverse: boolean,
+  scrollElement: ?Object,
+  children: Array<?React.Node> | ?React.Node,
+  className: string,
+};
+
+export default class InfiniteScroll extends React.Component<Props> {
+  static defaultProps = {
+    element: 'div',
+    hasMore: false,
+    initialLoad: true,
+    pageStart: 0,
+    threshold: 250,
+    useWindow: true,
+    isReverse: false,
+    scrollElement: null,
+    scrollComponent: null,
+  };
+
+  scrollListener: Function;
+  scrollComponent: Object;
+  pageLoaded: number;
+  _defaultLoader: React.Node;
+
+  constructor(props: Props) {
+    super(props);
+
+    this._defaultLoader = props.loader;
+    this.scrollListener = this.scrollListener.bind(this);
+  }
+
+  componentDidMount() {
+    this.pageLoaded = this.props.pageStart;
+    this.attachScrollListener();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const curr = this.props;
+
+    if (curr.isLoadingMore) {
+      return;
     }
 
-    componentWillUnmount() {
-      // if (!this.scrollBody) return;
-      // let node = this.scrollBody;
-      // if (node) {
-      // 	node.removeEventListener('scroll', this.handleScroll)
-      // }
+    if (
+      Array.isArray(prevProps.children) &&
+      Array.isArray(curr.children) &&
+      prevProps.children.length === curr.children.length
+    ) {
+      return;
     }
 
-    handleScroll = () => {};
+    if (
+      React.Children.toArray(curr.children).length === 1 &&
+      React.Children.toArray(curr.children)[0].type.name ===
+        'FlipMovePropConverter'
+    ) {
+      const currFlipMoveChildren = React.Children.toArray(curr.children)[0]
+        .props.children;
+      const prevFlipMoveChildren = React.Children.toArray(prevProps.children)[0]
+        .props.children;
```

Comment:
Would you mind adding a comment about wtf this code does? Why do you need to manually manipulate children?

Reply:
Yeah I'll add comments here; basically I need to avoid a refetch is the same number of children exist at this step; the problem is in the dashboard the `InfiniteScroll` has only one child: the `FlipMove` component. So this code is checking for the number of children that the flipmove component receives, which maps to the number of threads rendered. Overall I know this isn't perfect, but I was really struggling to find any way to prevent the component from calling two `fetchMore`s in a row with the same cursor ID, and this child length equality check does the job.